zerver/models.py

```diff
@@ -412,10 +412,54 @@ class Realm(models.Model):
     # Messages older than this message ID in the organization are inaccessible.
     first_visible_message_id: int = models.IntegerField(default=0)
 
-    # Valid org_types are {CORPORATE, COMMUNITY}
-    CORPORATE = 1
-    COMMUNITY = 2
-    org_type: int = models.PositiveSmallIntegerField(default=CORPORATE)
+    # Valid org_types
+    ORG_TYPES: Dict[str, Dict[str, Any]] = {
```

Comment:
I think we should add a value of `0` for "Unspecified", that we don't display as an option, but is the value that we use in the migration.  You might be want to encode that it's invisible here with a `hidden: true`, so the template code doesn't need to hardcode the value for excluding it from the dropdown.

I'd also recommend adding a `display_order` field here, so that we can just have a loop in the template displaying these sorted by that; that would mean we only have one place we need to list all the values.

Also, this can be `Dict[str, Union[int, str]]`

Reply:
As for the `display_order` field, we don't really need it since we will be displaying categories in alphabetical order. The code in the registration view takes care of that sorting.

As for changing the type annotation, specifying `Dict[str, Union[int, str]]` leads to mypy errors when I try to specify a constant ID as an optional parameter to a function like `Optional[int]=ORG_TYPES["business"]["id"]`. What does convention dictate here? Should we manually cast the ID, or change the `Optional` annotation to be a `Union` as well?

Reply:
@timabbott: Since we eventually decided to do semantic grouping for sorting the categories, please ignore my message about the `display_order` field, I will add it tomorrow! Thanks!

Reply:
I reworked the numbers here before merging to leave space between IDs and have the initial orders match.  While it doesn't matter for UI, it'll be nice for @alya when doing direct database queries to have the sort orders match.